compiler/rustc_hir_analysis/src/check/region.rs

```diff
@@ -887,3 +897,24 @@ pub fn region_scope_tree(tcx: TyCtxt<'_>, def_id: DefId) -> &ScopeTree {
 
     tcx.arena.alloc(scope_tree)
 }
+
+/// Check for reference of mutable static
+pub fn static_mut_ref(tcx: TyCtxt<'_>, stmts: &[Stmt<'_>]) {
+    for stmt in stmts {
+        if let hir::StmtKind::Local(loc) = stmt.kind
```

Comment:
This seems quite specific, would it trigger for any of these cases?

```
let ref x = X;
let (x, y) = (&X, &Y);
foo(&X);
```

I presume that we want to trigger whenever a reference is taken to a mutable static at all, not just when it is a simple assignment.